main/src/library/Foldable.flix

```diff
@@ -312,8 +312,8 @@ pub trait Foldable[t : Type -> Type] {
     ///
     /// Returns an immutable list of all the elements in `t` that satisfy the predicate `f`.
     ///
-    pub def filter(f: a -> Bool, t: t[a]): List[a] \ Foldable.Aef[t] =
-        Foldable.foldRight((x, acc) -> if (f(x)) x :: acc else acc, Nil, t)
+    pub def filter(f: a -> Bool \ ef, t: t[a]): List[a] \ {ef + Foldable.Aef[t]} =
```

Comment:
We usually only use `{ ... }` for sets of concrete effects. Here we are adding two effect sets together, so we prefer `ef + Foldable.Aef[t]`.